lib/getstats.js

```diff
@@ -52,10 +53,207 @@ function _getStats(peerConnection, options) {
+    { key: 'candidateType', type: 'string' },
+    { key: 'ip', type: 'string' },
+    { key: 'port', type: 'number' },
+    { key: 'priority', type: 'number' },
+    { key: 'protocol', type: 'string' },
+    { key: 'url', type: 'string' }
+  ];
+
+  var standardizedLocalCandidateStatsKeys = standardizedCandidateStatsKeys.concat([
+    { key: 'deleted', type: 'boolean' },
+    { key: 'relayProtocol', type: 'string' }
+  ]);
+
+  var standatdizedLocalCandidateStatsReport = activeLocalCandidateStats
+    ? standardizedLocalCandidateStatsKeys.reduce(function(report, keyInfo) {
+      report[keyInfo.key] = typeof activeLocalCandidateStats[keyInfo.key] === keyInfo.type
+        ? activeLocalCandidateStats[keyInfo.key]
+        : keyInfo.key === 'deleted' ? false : null;
+      return report;
+    }, {})
+    : null;
+
+  var standardizedRemoteCandidateStatsReport = activeRemoteCandidateStats
+    ? standardizedCandidateStatsKeys.reduce(function(report, keyInfo) {
+      report[keyInfo.key] = typeof activeRemoteCandidateStats[keyInfo.key] === keyInfo.type
+        ? activeRemoteCandidateStats[keyInfo.key]
+        : null;
+      return report;
+    }, {})
+    : null;
+
+  return [
+    { key: 'availableIncomingBitrate', type: 'number' },
+    { key: 'availableOutgoingBitrate', type: 'number' },
+    { key: 'bytesReceived', type: 'number' },
+    { key: 'bytesSent', type: 'number' },
+    { key: 'consentRequestsSent', type: 'number' },
+    { key: 'currentRoundTripTime', type: 'number' },
+    { key: 'lastPacketReceivedTimestamp', type: 'number' },
+    { key: 'lastPacketSentTimestamp', type: 'number' },
+    { key: 'nominated', type: 'boolean' },
+    { key: 'priority', type: 'number' },
+    { key: 'readable', type: 'boolean' },
+    { key: 'requestsReceived', type: 'number' },
+    { key: 'requestsSent', type: 'number' },
+    { key: 'responsesReceived', type: 'number' },
+    { key: 'responsesSent', type: 'number' },
+    { key: 'retransmissionsReceived', type: 'number' },
+    { key: 'retransmissionsSent', type: 'number' },
+    { key: 'state', type: 'string' },
+    { key: 'totalRoundTripTime', type: 'number' },
+    { key: 'transportId', type: 'string' },
+    { key: 'writable', type: 'boolean' }
+  ].reduce(function(report, keyInfo) {
+    report[keyInfo.key] = typeof activeCandidatePairStats[keyInfo.key] === keyInfo.type
+      ? activeCandidatePairStats[keyInfo.key]
+      : null;
+    return report;
+  }, {
+    localCandidate: standatdizedLocalCandidateStatsReport,
+    remoteCandidate: standardizedRemoteCandidateStatsReport
+  });
+}
+
+/**
+ * Standardize the active RTCIceCandidate pair's statistics in Firefox.
+ * @param {RTCStatsReport} stats
+ * @returns {?StandardizedActiveIceCandidatePairStatsReport}
+ */
+function standardizeFirefoxActiveIceCandidatePairStats(stats) {
+  stats = Object.values(stats);
+  var activeCandidatePairStats = stats.find(function(stat) {
+    return stat.type === 'candidatepair' && stat.nominated;
+  });
```

Comment:
Same comment as above.

Reply:
@markandrus In Firefox, `stats` is a JSON object, not a Map.